runtime/src/accounts_background_service.rs

```diff
@@ -184,6 +184,11 @@ impl SnapshotRequestHandler {
                     status_cache_slot_deltas,
                 } = snapshot_request;
 
+                if !snapshot_root_bank.is_startup_verification_complete() {
+                    // this cannot be trusted yet
+                    inc_new_counter_info!("handle_snapshot_requests-startup_verification_incomplete", 1);
+                }
```

Comment:
yes, it should never get hit.
assert kills the validator?
it is an assumption so I thought I'd at last log if it is happening.
We don't know it is wrong - we just can't know yet.
In the future or the present perhaps there are other ways to request a snapshot that circumvent the check in bank forks.
Do you think an assert is better?

Reply:
Yes, I think an assert here will kill the validator. I like asserts when it's a programmer bug, because then it blows up early and loudly.

> We don't know it is wrong - we just can't know yet.

I'm not sure what this part means.

> In the future or the present perhaps there are other ways to request a snapshot that circumvent the check in bank forks.

Right now there are no other ways (other than tests) to send a snapshot request. If this changes in the future, the assert could be replaced as well. Do you anticipate more places sending snapshot requests?

> Do you think an assert is better?

In this case I do. However not strongly tied to it, so I'll defer it back to your call.

Reply:
ok, changed to an assert.